operators/benches/cache_concurrent.rs

```diff
@@ -0,0 +1,219 @@
+    raster::{Grid, RasterTile2D},
+    util::test::TestDefault,
+};
+use geoengine_operators::{engine::CanonicOperatorName, pro::cache::tile_cache::TileCache};
+use serde_json::json;
+
+enum Measurement {
+    Read(ReadMeasurement),
+    Write(WriteMeasurement),
+}
+
+struct ReadMeasurement {
+    read_query_ms: u128,
+}
+
+struct WriteMeasurement {
+    insert_query_ms: u128,
+    insert_tile_ms: u128,
+    finish_query_ms: u128,
+}
+
+async fn write_cache(tile_cache: &TileCache, op_name: CanonicOperatorName) -> WriteMeasurement {
+    let tile = RasterTile2D::<u8> {
+        time: TimeInterval::new_unchecked(1, 1),
+        tile_position: [-1, 0].into(),
+        global_geo_transform: TestDefault::test_default(),
+        grid_array: Grid::new([3, 2].into(), vec![1, 2, 3, 4, 5, 6])
+            .unwrap()
+            .into(),
+        properties: RasterProperties::default(),
+    };
+
+    let start = std::time::Instant::now();
+
+    let query_id = tile_cache
+        .insert_query::<u8>(op_name.clone(), &query_rect())
+        .await;
+
+    let insert_query_s = start.elapsed().as_millis();
+
+    let start = std::time::Instant::now();
+
+    tile_cache
+        .insert_tile(op_name.clone(), query_id, tile)
+        .await
+        .unwrap();
+
+    let insert_tile_s = start.elapsed().as_millis();
+
+    let start = std::time::Instant::now();
+
+    tile_cache
+        .finish_query(op_name.clone(), query_id)
+        .await
+        .unwrap();
+
+    let finish_query_s = start.elapsed().as_millis();
+
+    WriteMeasurement {
+        insert_query_ms: insert_query_s,
+        insert_tile_ms: insert_tile_s,
+        finish_query_ms: finish_query_s,
+    }
+}
+
+async fn read_cache(tile_cache: &TileCache) -> ReadMeasurement {
+    let op = op(0);
+    let query = query_rect();
+
+    let start = std::time::Instant::now();
+
+    tile_cache.query_cache::<u8>(op, &query).await;
+
+    let read_query_s = start.elapsed().as_millis();
+
+    ReadMeasurement {
+        read_query_ms: read_query_s,
+    }
+}
+
+fn query_rect() -> RasterQueryRectangle {
+    RasterQueryRectangle {
+        spatial_bounds: SpatialPartition2D::new_unchecked((-180., 90.).into(), (180., -90.).into()),
+        time_interval: TimeInterval::new_instant(DateTime::new_utc(2014, 3, 1, 0, 0, 0)).unwrap(),
+        spatial_resolution: SpatialResolution::one(),
+    }
+}
+
+fn op(idx: usize) -> CanonicOperatorName {
+    CanonicOperatorName::new(json!({
+        "type": "GdalSource",
+        "params": {
+            "data": idx
+        }
+    }))
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Operation {
+    Read,
+    Write,
+}
+
+async fn cache_access(
+    tile_cache: Arc<TileCache>,
+    op_no: usize,
+    operation: Operation,
+) -> Measurement {
+    match operation {
+        Operation::Read => Measurement::Read(read_cache(&tile_cache).await),
+        Operation::Write => Measurement::Write(write_cache(&tile_cache, op(op_no)).await),
+    }
```

Comment:
well the plots clearly show that it's doing something, right?

Reply:
Yeah, but it could be different if it could not optimize certain inputs or outputs.
It's not like that the benchmark is ready in 0s or computes everything like in a real scenario.